packages/sdk/src/apiClient/certificateAuthProvider.ts

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AxiosRequestConfig } from "axios";
+import { Agent } from "https";
+import { SecureContextOptions } from "tls";
+import { AuthProvider } from "./authProvider";
+import { ErrorWithCode, ErrorCode, ErrorMessage } from "../core/errors";
+import { formatString } from "../util/utils";
+
+/**
+ * Provider that handles Certificate authentication
+ *
+ * @beta
+ */
+
+export class CertificateAuthProvider implements AuthProvider {
+  private certOption: SecureContextOptions;
+
+  /**
+   *
+   * @param { SecureContextOptions } certOption - information about the cert used in http requests
+   *
+   * @beta
+   */
+  constructor(certOption: SecureContextOptions) {
+    this.certOption = certOption;
```

Comment:
will the cerOption will be empty object?

Reply:
It's hard to validate whether the certOption is valid. So I only validates user input in `createPemCertOption` and `createPfxCertOption`, since we have clear scenario for these 2 helper functions.

If advanced user wants to pass their own certOption (which is not generated by our helper function) here, they need to ensure the option is correct. And we rely on Axios to find problems in the certOption.